video/video.ino

```diff
@@ -96,6 +97,10 @@ void loop() {
 
 	while (true) {
 		if (terminalMode) {
+			if (exitTerminal) {
+				switchTerminalMode(false);
+				continue;

```

Comment:
this was the secret sauce to getting this to work - using a flag to indicate that we want to edit terminal mode.

calling `switchTerminalMode` directly inside the terminal delegate functions was resulting in exceptions being thrown and the VDP crashing.  that was almost certainly a multi-threading related issue.